app/server/src/routes/userProfile/userProfile.controller.js

```diff
@@ -0,0 +1,59 @@
+const userProfileModel = require("../../models/userProfile/userProfile.model");
+const personalInfoModel = require("../../models/personalInfo/personalInfo.model");
+const badgeModel = require("../../models/badges/badges.model");
+const mongoose = require("mongoose");
+
+const UserProfileController = {
+    
+    postCreate: async function (req,res) {
+
+        try{
+            const {name, surname, email} = req.query
+            const [idProfile, profile] = await userProfileModel.createUserProfile(name, surname, email, Date.now())
+            const [idInfo, info] = await personalInfoModel.createPersonalInfo(idProfile)
+            profile.personal_info = idInfo
+            profile.save()
+            res.status(201).send({"message": "Created user profile successfully"})
+
+        }   
+        catch (e) {
+            res.status(400).send({"error": e})
+        }
+    }, 
+
+    getProfile: async function (req,res) {
```

Comment:
Again to get rid of userprofile, use the coming `id` to get `user` not userprofile, can get user via user.model getuserbyid
The rest is pretty much similar. To help you with acquiring personal_info in user and badges in personal_info
I can suggest you to use populate function,
basically works something like this
```jsx
user.populate({
      path: 'personal_info',
      populate: {
        path: 'badges',
      }
    })
    .exec()
```
this would basically open up the id's into their related data

Reply:
According to the API specifications, I think we should create "getUser" endpoint in userController since it requests more information than what's in personalInfo.

Reply:
Yes i agree, but we have some wording problems in class diagrams and api specs :D I think this can stay as getProfile since it would probably be used for filling data in profile page, which requires not only personal info but name surname email etc. To sum up, personal info != profile in our case. Profile is personal info + user data for now.